lib/queuePopulator/IngestionProducer.js

```diff
@@ -383,6 +401,28 @@ class IngestionProducer {
         }
         return done(null, []);
     }
+
+    /**
+     * Get bucket cseq
+     * @param {string} bucket - bucket name
+     * @param {function} done - callback(err, cseq) where `cseq` is an integer
+     * @return {undefined}
+     */
+    _getBucketCseq(bucket, done) {
+        return this._ringReader.getBucketCseq({ Bucket: bucket },
+        (err, data) => {
+            if (err) {
+                this.log.error('error getting bucket cseq', {
+                    method: 'IngestionProducer._getBucketCseq',
+                    error: err,
+                    bucket,
+                });
+                return done(err);
+            }
+            // cseq returned by all nodes. Just return the first node response
+            return done(null, data[0].cseq);
```

Comment:
It might be safe to check that `data.length > 0` and `data[0]` is not undefined.

Reply:
ty, updated :)